src/main/java/frc/trigon/robot/subsystems/turret/simulationturret/SimulationTurretIO.java

```diff
@@ -0,0 +1,59 @@
+package frc.trigon.robot.subsystems.turret.simulationturret;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.subsystems.turret.TurretIO;
+import frc.trigon.robot.subsystems.turret.TurretInputsAutoLogged;
+import frc.trigon.robot.utilities.Conversions;
+
+public class SimulationTurretIO extends TurretIO {
+    private final DCMotorSim motor = SimulationTurretConstants.MOTOR;
+
+    private double motorVoltage;
```

Comment:
You're not updating this variable.

Reply:
Are you sure it isn't? On my side it looks completely updated.